url4fdf79ae2c_beckchr_juel_tgz-pJ8-de_odysseus_el_samples_xml_sax_XMELFilterJ8/src/de/odysseus/el/misc/BooleanOperations.java

```diff
@@ -0,0 +1,191 @@
+		}
+		if (SIMPLE_INTEGER_TYPES.contains(t1) || SIMPLE_INTEGER_TYPES.contains(t2)) {
+			return converter.convert(o1, Long.class) < converter.convert(o2, Long.class);
+		}
+		if (t1 == String.class || t2 == String.class) {
+			return converter.convert(o1, String.class).compareTo(converter.convert(o2, String.class)) < 0;
+		}
+		if (o1 instanceof Comparable) {
+			return ((Comparable)o1).compareTo(o2) < 0;
+		}
+		if (o2 instanceof Comparable) {
+			return ((Comparable)o2).compareTo(o1) > 0;
+		}
+		throw new ELException(LocalMessages.get("error.compare.types", o1.getClass(), o2.getClass()));
+	}
+
+	@SuppressWarnings("unchecked")
+	private static final boolean gt0(TypeConverter converter, Object o1, Object o2) {		
+		Class<?> t1 = o1.getClass();
+		Class<?> t2 = o2.getClass();
+		if (BigDecimal.class.isAssignableFrom(t1) || BigDecimal.class.isAssignableFrom(t2)) {
+			return converter.convert(o1, BigDecimal.class).compareTo(converter.convert(o2, BigDecimal.class)) > 0;
+		}
+		if (SIMPLE_FLOAT_TYPES.contains(t1) || SIMPLE_FLOAT_TYPES.contains(t2)) {
+			return converter.convert(o1, Double.class) > converter.convert(o2, Double.class);
+		}
+		if (BigInteger.class.isAssignableFrom(t1) || BigInteger.class.isAssignableFrom(t2)) {
+			return converter.convert(o1, BigInteger.class).compareTo(converter.convert(o2, BigInteger.class)) > 0;
+		}
+		if (SIMPLE_INTEGER_TYPES.contains(t1) || SIMPLE_INTEGER_TYPES.contains(t2)) {
+			return converter.convert(o1, Long.class) > converter.convert(o2, Long.class);
+		}
+		if (t1 == String.class || t2 == String.class) {
+			return converter.convert(o1, String.class).compareTo(converter.convert(o2, String.class)) > 0;
+		}
+		if (o1 instanceof Comparable) {
+			return ((Comparable)o1).compareTo(o2) > 0;
+		}
+		if (o2 instanceof Comparable) {
+			return ((Comparable)o2).compareTo(o1) < 0;
+		}
+		throw new ELException(LocalMessages.get("error.compare.types", o1.getClass(), o2.getClass()));
+	}
+
+	public static final boolean lt(TypeConverter converter, Object o1, Object o2) {
+		if (o1 == o2) {
+			return false;
+		}
+		if (o1 == null || o2 == null) {
+			return false;
+		}
+		return lt0(converter, o1, o2);
+	}
+
+	public static final boolean gt(TypeConverter converter, Object o1, Object o2) {
+		if (o1 == o2) {
+			return false;
+		}
+		if (o1 == null || o2 == null) {
+			return false;
+		}
+		return gt0(converter, o1, o2);
+	}
+
+	public static final boolean ge(TypeConverter converter, Object o1, Object o2) {
+		if (o1 == o2) {
+			return true;
+		}
+		if (o1 == null || o2 == null) {
+			return false;
+		}
+		return !lt0(converter, o1, o2);
+	}
+
+	public static final boolean le(TypeConverter converter, Object o1, Object o2) {
+		if (o1 == o2) {
+			return true;
+		}
+		if (o1 == null || o2 == null) {
+			return false;
+		}
+		return !gt0(converter, o1, o2);
+	}
+
+	public static final boolean eq(TypeConverter converter, Object o1, Object o2) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.